hack/bundle-gen.py

```diff
@@ -0,0 +1,410 @@
+#!/usr/bin/env python
+
+import argparse
+import datetime
+import git
+import importlib
+import json
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+import urllib3
+import yaml
+
+
+# HIVE_VERSION_PREFIX is the prefix of the Hive version that will be constructed by this script
+# the version used for images and bundles will be "{prefix}.{number of commits}-{git hash}" eg. "v1.2.3187-18827f6"
+HIVE_VERSION_PREFIX = "v1.2"
+
+# HIVE_COMMIT_HASH_DEFAULT is the default hive commit hash referenced by this script for building
+# the hive image, generating the version, etc.
+HIVE_COMMIT_HASH_DEFAULT = 'HEAD'
+
+HIVE_REPO_DIR_DEFAULT = '/home/{}/go/src/github.com/openshift/hive'.format(os.environ['USER'])
+
+# Hive dir within both:
+# https://github.com/redhat-openshift-ecosystem/community-operators-prod
+# https://github.com/k8s-operatorhub/community-operators
+HIVE_SUB_DIR = 'operators/hive-operator'
+
+REGISTRY_AUTH_FILE_DEFAULT = '/home/{}/.docker/config.json'.format(os.environ['USER'])
+
+COMMUNITY_OPERATORS_HIVE_PKG_URL = 'https://raw.githubusercontent.com/redhat-openshift-ecosystem/community-operators-prod/main/operators/hive-operator/hive.package.yaml'
+
+CHANNEL_DEFAULT = 'alpha'
+
+CSV_DEFAULT = 'operatorhub'
+
+SUBPROCESS_REDIRECT = subprocess.DEVNULL
+
+def get_params():
+    parser = argparse.ArgumentParser(description='')
+    parser.add_argument('--commit-hash',
+                        default=HIVE_COMMIT_HASH_DEFAULT,
+                        help='Git commit ID associated with new version')
+    parser.add_argument('--hive-repo-dir',
+                        default=HIVE_REPO_DIR_DEFAULT,
+                        help='Path to root of repo to checkout/build images from')
+    parser.add_argument('--registry-auth-file',
+                        default=REGISTRY_AUTH_FILE_DEFAULT,
+                        help='Path to registry auth file')
+    parser.add_argument('--channel',
+                        default=[CHANNEL_DEFAULT],
+                        action='append',
+                        help='Update channel in OLM package to new version')
```

Comment:
I think we shouldn't need this arg. If we can't detect what channel(s) to update, something is wrong. (Unless this script is also going to support fast-forwarding OCM branches, which I think is probably too much.)